output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/Main.java

```diff
@@ -0,0 +1,191 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.outputcomparator.cli;
+
+import com.beust.jcommander.JCommander;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import org.mobilitydata.gtfsvalidator.model.ValidationReport;
+import org.mobilitydata.gtfsvalidator.outputcomparator.io.NoticeComparisonReport;
+import org.mobilitydata.gtfsvalidator.outputcomparator.model.SourceUrlContainer;
+
+public class Main {
+
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final String ACCEPTANCE_REPORT_JSON = "acceptance_report.json";
+  private static final int IO_EXCEPTION_EXIT_CODE = 1;
+  private static final int INVALID_NEW_RULE_EXIT_CODE = 2;
+  private static final Gson GSON =
+      new GsonBuilder().serializeNulls().disableHtmlEscaping().create();
+
+  public static void main(String[] argv) {
+    Arguments args = new Arguments();
+    new JCommander(args).parse(argv);
+    List<File> reportDirectory = null;
+    try {
+      reportDirectory =
+          Files.list(Paths.get(args.getReportDirectory()))
+              .map(Path::toFile)
+              .collect(Collectors.toList());
+      if (reportDirectory.isEmpty()) {
+        logger.atSevere().log(
+            "Specified directory is empty, cannot generate acceptance tests report.");
+        return;
+      }
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException);
+      System.exit(IO_EXCEPTION_EXIT_CODE);
+    }
+    int badDatasetCount = 0;
+    List<File> reportDirs =
+        reportDirectory.stream().filter(File::isDirectory).collect(Collectors.toList());
+    Map<String, NoticeComparisonReport> acceptanceTestReportMap = new TreeMap<>();
+
+    SourceUrlContainer sourceUrlContainer = null;
+    try {
+      sourceUrlContainer = new SourceUrlContainer(args.getSourceUrlPath());
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException);
+      System.exit(IO_EXCEPTION_EXIT_CODE);
+    }
+
+    for (File file : reportDirs) {
+      Path referenceReportPath = file.toPath().resolve(args.getReferenceValidationReportName());
+      Path latestReportPath = file.toPath().resolve(args.getLatestValidationReportName());
+      if (!Path.of(referenceReportPath.toString()).toFile().isFile()
+          || !Path.of(latestReportPath.toString()).toFile().isFile()) {
+        continue;
```

Comment:
I'm still concerned that we're easily skipping non-existing files or files that failed to to be read. 

Re GitHub created folders: 
It looks like we consider `file.getName` to be `sourceId` this way we can check if this is valid source id and identify whether it's GitHub specific folder or not.

Re ignoring corrupted files or non-existence: 
we should guard against cases when we allow bad pull request to go through because some of the files (or all of them) were missed or failed on read. I can think of two options
 * we can consider missed files or corrupted files to have an error in them  and include them into `badDatasetCount` and report
 *  log every bad source like this count number of failed sources and if this number is greater than some value we consider test to be invalid
There should be some other ways as well, what do you think?

Reply:
> There should be some other ways as well, what do you think?

👍🏾 -For clarity, I would separate `badDatasetCount` from the number of corrupted sources by using `corruptedSourcesCount`.

https://github.com/MobilityData/gtfs-validator/pull/848/commits/7a433434cd15c47dd0ce999d6b79237c9ca65641 provides the following changes:
1. check if `file.getName` is a valid sourceId using `SourceUrlContainer`
1. count the number of corrupted sources: if said number is greater than or equal to the threshold passed by user then the acceptance test fails. As of now the threshold is set to 2%, the acceptance test report is still available in this case.
2. A message is emitted in the previous case with information regarding the sources that are corrupted/could not be read.